Cleanup

alonsogarciapablo opened the pull request:
In preparation for #466, in this PR:
- I've created `cdb.core.util` and moved some utility methods that were included in `layer_definition.js`.
- I've also re-organized some methods in `layer_definition.js` and organized tests a bit.
- I've moved 'Sublayer' to it's own file.
- I've moved some specs to the right folder.

@javisantana PTAL.

PS: I still need to add `util.js` to grunt tasks.

javisantana commented:
don't forget to add util.js to cartodb.js file list (it's the way to add files to the build, probably we need to change that)

Apart from that It's perfect

alonsogarciapablo commented:
retest this please

alonsogarciapablo commented:
retest this please

alonsogarciapablo commented:
@javisantana I [added util.js to the list of files in src/cartodb.js](https://github.com/CartoDB/cartodb.js/commit/b6ddb85fa3e472038d144fbe4731723091c613ea). Do I need to add it anywhere else? Otherwise I will merge this.

alonsogarciapablo commented:
@javisantana I've added some more commits with some more cleanup. PTAL. I've updated the description with the changes included in this PR. I'd like to merge this before continuing working on #466,  so that we don't end up having a massive PR. Thanks!

alonsogarciapablo commented on src/geo/layer_definition.js:
```diff
@@ -661,242 +456,169 @@ Map.prototype = {
+  for (var i = 0; i < sublayers.length; ++i) {
+    layer_definition.layers.push({
+      type: 'cartodb',
+      options: sublayers[i]
```
Hey @javisantana, I think there's something wrong here. We're adding adding everything in `sublayers[i]` to the `options` attribute. That's generating something like:
```
{
  type: 'cartodb',
  options: {
    sql: '...',
    cartocss: '...',
  }
}
```

But according to the spec, it should be something like:

```
{
  type: 'cartodb',
  sql: '...',
  cartocss: '...',
  options: {
    ...
  }
}
```

Also, methods like `Sublayer#setSQL` are setting those attribute inside of the `options` attribute, which seems wrong to me.

What do you think?

javisantana replied:
what spec?

alonsogarciapablo replied:
I was referring to MapConfig, but it's actually right :)
